### ISSUE TITLE
What has changed in renderPlot/plotOutput?

### ISSUE THREAD

**Issue by stla** [rstudio/shiny#3854]
Hello,

Previously, on december 2022, my package [swiper](https://github.com/stla/swipeR) worked fine. In particular the cube effect didn't have any issue:

![swipeR-shiny](https://github.com/rstudio/shiny/assets/4466543/a33b8420-fece-408c-afe4-8a54ca9feb2c)

But now it does not work fine: the dimensions of the second and the fourth graphic of the cube are totally screwed up. They became normal after a resize.

So I'm wondering what has changed in Shiny since december? I tried the option `fill` of `plotOutput`, this is the only new thing I identified, but it didn't solve the issue.

The cube works fine with static images. This is a problem with `renderPlot` or `plotOutput`, I don't know.

**Comment by gadenbuie** [comment]
Hi @stla thanks for opening this issue. Your intuition may be right about the fill option of `plotOutput`, or it could be something else. Could you please include a reprex that demonstrates the behavior you're seeing? (The examples in the swiper README all worked for me, but they don't include Shiny or `plotOutput()`.)

**Comment by gadenbuie** [comment]
Thanks @stla. Here's a quick analysis. In general, Shiny measures the contents of the plot container and then redraws the plot to fit that container. When the page loads, the first and fourth pane are at the front and back of the cube, so they're correctly sized. The two side panes have very small widths, so the plots render at an unusual size.

The easiest way to avoid this is to render your plots at a static size by setting `width` and `height` in `renderPlot()`. In the case of your example app, I'd find a reasonable height/width combination that looks good for all of the plots and then I'd probably use CSS on the parent container to constrain the width (instead of setting `width = 70%`).

The other option you have is to dispatch a resize event on `window`. Shiny listens to these events and recalculates plot sizes when that happens. You could try adding `$(window).trigger("resize")` to the `onclick` handlers in your action buttons, for example.

In general, @cpsievert and I have been talking about having Shiny automatically resize plots when their container sizes change. This is a good example for us to use in motivating that change, so thank you for bringing it up!

**Comment by stla** [comment]
Thanks, I will try to set the dimensions in pixels.

I already tried to trigger the `resize` event, this works but this is ugly, this flickers.

**Comment by gadenbuie** [comment]
> I already tried to trigger the `resize` event, this works but this is ugly, this flickers.

Agreed. The transitions should be much better with an htmlwidget like plotly, though.

**Comment by cpsievert** [comment]
> other option you have is to dispatch a resize event on window

After #3682, this should no longer be needed.
